syntax = "proto3";

package splitlearning;

message ClientToServer {
    repeated float activations = 1;
    repeated int32 labels      = 2;
    int32 batch_size           = 3;
    int32 client_id            = 4;
}

message ServerToClient {
    repeated float gradients = 1;
    float loss               = 2;
    float server_time        = 3;
    float acc                = 4;
}

service SplitLearning {
    rpc SendClientActivations(ClientToServer) returns (ServerToClient) {}
}
